llmfoundry/data/finetuning/dataloader.py

```diff
@@ -46,7 +46,7 @@
 
 def build_finetuning_dataloader(
     tokenizer: PreTrainedTokenizerBase,
-    device_batch_size: int,
+    device_batch_size: Union[int, float],
```

Comment:
Can we add a unit test for non-integer batch size?

Reply:
discussing offline